plugins/serializers/carbon2/carbon2.go

```diff
@@ -0,0 +1,66 @@
+package carbon2
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"strconv"
+	"strings"
+)
+
+type serializer struct {
+}
+
+func NewSerializer() (*serializer, error) {
+	s := &serializer{}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	return []byte(s.createObject(metric)), nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+	var batch bytes.Buffer
+	for _, metric := range metrics {
+		batch.WriteString(s.createObject(metric))
+	}
+	return []byte(batch.String()), nil
```

Comment:
Since the bytes.Buffer is no longer modified, you can return return batch.Bytes() to avoid a copy.